File: 1502-construct-k-palindrome-strings/construct-k-palindrome-strings.java
class Solution {
    public boolean canConstruct(String s, int k) {
        if (k > s.length()) return false;
        int oddCount = 0;
        int[] freq = new int[26];
        for (char c : s.toCharArray()) {
            freq[c - 'a']++;
        }
        for (int count : freq) {
            if (count % 2 != 0) oddCount++;
        }
        return k >= oddCount;
    }
}

File: 1502-construct-k-palindrome-strings/construct-k-palindrome-strings.py
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        if k > len(s):
            return False
        odd_count = 0
        for char in set(s):
            if s.count(char) % 2 != 0:
                odd_count += 1
        return k >= odd_count